### ISSUE TITLE
Bedrock Tool Call Broken

### ISSUE THREAD

**Issue by Amorim33** [vercel/ai#3446]
### Description

Thank you for making this fantastic code open-source! ✨

After updating `@ai-sdk/amazon-bedrock` to the latest version, I've noticed that tool-calling functionality has stopped working. To investigate further, I cloned the repository and tried running the example files. Specifically, the `amazon-bedrock-tool-call` example doesn’t seem to function as expected. I’ve tested it with both the Claude 3.5 Sonnet model and a custom Claude 3.5 Sonnet V2 that includes Amazon Inference Profiles.

I suspect that PR [#3353](https://github.com/vercel/ai/pull/3353) might have introduced this regression.

#### Steps to Reproduce

1. Clone the repository and navigate to the examples directory.
2. Run the `amazon-bedrock-tool-call` example with the latest `@ai-sdk/amazon-bedrock` version.
3. Observe that tool calling does not work as expected when using the Claude 3.5 Sonnet model or a custom Claude 3.5 Sonnet V2 (did not tested with others).

#### Raw Logs
```
{
  "text": "I'm sorry, but I don't have access to real-time weather information or current conditions in San Francisco. Weather can change frequently, so it's best to check a reliable weather website or app for the most up-to-date information.\n\nAs for attractions in San Francisco, here are some popular ones you might consider visiting:\n\n1. Golden Gate Bridge\n2. Alcatraz Island\n3. Fisherman's Wharf\n4. Chinatown\n5. Pier 39\n6. Golden Gate Park\n7. Lombard Street\n8. Cable Cars\n9. Palace of Fine Arts\n10. Ghirardelli Square\n11. California Academy of Sciences\n12. San Francisco Museum of Modern Art (SFMOMA)\n13. Exploratorium\n14. Twin Peaks\n15. The Painted Ladies (Victorian houses)\n\nThe best attractions for you will depend on your interests, the time you have available, and the season of your visit. It's always a good idea to check the official websites of these attractions for current operating hours, ticket information, and any potential closures or restrictions.",
  "toolCalls": [],
  "toolResults": [],
  "finishReason": "stop",
  "usage": {
    "completionTokens": 243,
    "promptTokens": 21,
    "totalTokens": 264
  },
  "warnings": [],
  "request": {},
  "response": {
    "id": "aitxt-mXUBIDGLdyOsNHPwMXBjTE5a",
    "timestamp": "2024-11-01T13:04:52.106Z",
    "modelId": "anthropic.claude-3-5-sonnet-20240620-v1:0",
    "messages": [
      {
        "role": "assistant",
        "content": [
          {
            "type": "text",
            "text": "I'm sorry, but I don't have access to real-time weather information or current conditions in San Francisco. Weather can change frequently, so it's best to check a reliable weather website or app for the most up-to-date information.\n\nAs for attractions in San Francisco, here are some popular ones you might consider visiting:\n\n1. Golden Gate Bridge\n2. Alcatraz Island\n3. Fisherman's Wharf\n4. Chinatown\n5. Pier 39\n6. Golden Gate Park\n7. Lombard Street\n8. Cable Cars\n9. Palace of Fine Arts\n10. Ghirardelli Square\n11. California Academy of Sciences\n12. San Francisco Museum of Modern Art (SFMOMA)\n13. Exploratorium\n14. Twin Peaks\n15. The Painted Ladies (Victorian houses)\n\nThe best attractions for you will depend on your interests, the time you have available, and the season of your visit. It's always a good idea to check the official websites of these attractions for current operating hours, ticket information, and any potential closures or restrictions."
          }
        ]
      }
    ]
  },
  "responseMessages": [
    {
      "role": "assistant",
      "content": [
        {
          "type": "text",
          "text": "I'm sorry, but I don't have access to real-time weather information or current conditions in San Francisco. Weather can change frequently, so it's best to check a reliable weather website or app for the most up-to-date information.\n\nAs for attractions in San Francisco, here are some popular ones you might consider visiting:\n\n1. Golden Gate Bridge\n2. Alcatraz Island\n3. Fisherman's Wharf\n4. Chinatown\n5. Pier 39\n6. Golden Gate Park\n7. Lombard Street\n8. Cable Cars\n9. Palace of Fine Arts\n10. Ghirardelli Square\n11. California Academy of Sciences\n12. San Francisco Museum of Modern Art (SFMOMA)\n13. Exploratorium\n14. Twin Peaks\n15. The Painted Ladies (Victorian houses)\n\nThe best attractions for you will depend on your interests, the time you have available, and the season of your visit. It's always a good idea to check the official websites of these attractions for current operating hours, ticket information, and any potential closures or restrictions."
        }
      ]
    }
  ],
  "roundtrips": [
    {
      "stepType": "initial",
      "text": "I'm sorry, but I don't have access to real-time weather information or current conditions in San Francisco. Weather can change frequently, so it's best to check a reliable weather website or app for the most up-to-date information.\n\nAs for attractions in San Francisco, here are some popular ones you might consider visiting:\n\n1. Golden Gate Bridge\n2. Alcatraz Island\n3. Fisherman's Wharf\n4. Chinatown\n5. Pier 39\n6. Golden Gate Park\n7. Lombard Street\n8. Cable Cars\n9. Palace of Fine Arts\n10. Ghirardelli Square\n11. California Academy of Sciences\n12. San Francisco Museum of Modern Art (SFMOMA)\n13. Exploratorium\n14. Twin Peaks\n15. The Painted Ladies (Victorian houses)\n\nThe best attractions for you will depend on your interests, the time you have available, and the season of your visit. It's always a good idea to check the official websites of these attractions for current operating hours, ticket information, and any potential closures or restrictions.",
      "toolCalls": [],
      "toolResults": [],
      "finishReason": "stop",
      "usage": {
        "promptTokens": 21,
        "completionTokens": 243,
        "totalTokens": 264
      },
      "warnings": [],
      "request": {},
      "response": {
        "id": "aitxt-mXUBIDGLdyOsNHPwMXBjTE5a",
        "timestamp": "2024-11-01T13:04:52.106Z",
        "modelId": "anthropic.claude-3-5-sonnet-20240620-v1:0",
        "messages": [
          {
            "role": "assistant",
            "content": [
              {
                "type": "text",
                "text": "I'm sorry, but I don't have access to real-time weather information or current conditions in San Francisco. Weather can change frequently, so it's best to check a reliable weather website or app for the most up-to-date information.\n\nAs for attractions in San Francisco, here are some popular ones you might consider visiting:\n\n1. Golden Gate Bridge\n2. Alcatraz Island\n3. Fisherman's Wharf\n4. Chinatown\n5. Pier 39\n6. Golden Gate Park\n7. Lombard Street\n8. Cable Cars\n9. Palace of Fine Arts\n10. Ghirardelli Square\n11. California Academy of Sciences\n12. San Francisco Museum of Modern Art (SFMOMA)\n13. Exploratorium\n14. Twin Peaks\n15. The Painted Ladies (Victorian houses)\n\nThe best attractions for you will depend on your interests, the time you have available, and the season of your visit. It's always a good idea to check the official websites of these attractions for current operating hours, ticket information, and any potential closures or restrictions."
              }
            ]
          }
        ]
      },
      "isContinued": false
    }
  ],
  "steps": [
    {
      "stepType": "initial",
      "text": "I'm sorry, but I don't have access to real-time weather information or current conditions in San Francisco. Weather can change frequently, so it's best to check a reliable weather website or app for the most up-to-date information.\n\nAs for attractions in San Francisco, here are some popular ones you might consider visiting:\n\n1. Golden Gate Bridge\n2. Alcatraz Island\n3. Fisherman's Wharf\n4. Chinatown\n5. Pier 39\n6. Golden Gate Park\n7. Lombard Street\n8. Cable Cars\n9. Palace of Fine Arts\n10. Ghirardelli Square\n11. California Academy of Sciences\n12. San Francisco Museum of Modern Art (SFMOMA)\n13. Exploratorium\n14. Twin Peaks\n15. The Painted Ladies (Victorian houses)\n\nThe best attractions for you will depend on your interests, the time you have available, and the season of your visit. It's always a good idea to check the official websites of these attractions for current operating hours, ticket information, and any potential closures or restrictions.",
      "toolCalls": [],
      "toolResults": [],
      "finishReason": "stop",
      "usage": {
        "promptTokens": 21,
        "completionTokens": 243,
        "totalTokens": 264
      },
      "warnings": [],
      "request": {},
      "response": {
        "id": "aitxt-mXUBIDGLdyOsNHPwMXBjTE5a",
        "timestamp": "2024-11-01T13:04:52.106Z",
        "modelId": "anthropic.claude-3-5-sonnet-20240620-v1:0",
        "messages": [
          {
            "role": "assistant",
            "content": [
              {
                "type": "text",
                "text": "I'm sorry, but I don't have access to real-time weather information or current conditions in San Francisco. Weather can change frequently, so it's best to check a reliable weather website or app for the most up-to-date information.\n\nAs for attractions in San Francisco, here are some popular ones you might consider visiting:\n\n1. Golden Gate Bridge\n2. Alcatraz Island\n3. Fisherman's Wharf\n4. Chinatown\n5. Pier 39\n6. Golden Gate Park\n7. Lombard Street\n8. Cable Cars\n9. Palace of Fine Arts\n10. Ghirardelli Square\n11. California Academy of Sciences\n12. San Francisco Museum of Modern Art (SFMOMA)\n13. Exploratorium\n14. Twin Peaks\n15. The Painted Ladies (Victorian houses)\n\nThe best attractions for you will depend on your interests, the time you have available, and the season of your visit. It's always a good idea to check the official websites of these attractions for current operating hours, ticket information, and any potential closures or restrictions."
              }
            ]
          }
        ]
      },
      "isContinued": false
    }
  ],
  "rawResponse": {}
}
```

### Code example
```ts
import { generateText, tool } from 'ai';
import 'dotenv/config';
import { z } from 'zod';
import { weatherTool } from '../tools/weather-tool';
import { bedrock } from '@ai-sdk/amazon-bedrock';

async function main() {
  const result = await generateText({
    model: bedrock('anthropic.claude-3-5-sonnet-20240620-v1:0'),
    maxTokens: 512,
    tools: {
      weather: weatherTool,
      cityAttractions: tool({
        parameters: z.object({ city: z.string() }),
      }),
    },
    prompt:
      'What is the weather in San Francisco and what attractions should I visit?',
  });

  // typed tool calls:
  for (const toolCall of result.toolCalls) {
    switch (toolCall.toolName) {
      case 'cityAttractions': {
        toolCall.args.city; // string
        break;
      }

      case 'weather': {
        toolCall.args.location; // string
        break;
      }
    }
  }

  // typed tool results for tools with execute method:
  for (const toolResult of result.toolResults) {
    switch (toolResult.toolName) {
      // NOT AVAILABLE (NO EXECUTE METHOD)
      // case 'cityAttractions': {
      //   toolResult.args.city; // string
      //   toolResult.result;
      //   break;
      // }

      case 'weather': {
        toolResult.args.location; // string
        toolResult.result.location; // string
        toolResult.result.temperature; // number
        break;
      }
    }
  }

  console.log(JSON.stringify(result, null, 2));
}

main().catch(console.error);
```

### Additional context

Thank you again!
If this aligns with the library goals, I would be stoked to submit a PR!🙏

**Comment by lgrammel** [comment]
I just tested and it works for me. Have you updated all ai sdk packages to the latest version (or rebuilt the repository if you are using it directly via `pnpm build` in root)?

**Comment by Amorim33** [comment]
I'm up-to-date with main
![image](https://github.com/user-attachments/assets/9f7a1bc2-45bd-480e-a7bf-47c61b40319c)

I've ran install and build
![image](https://github.com/user-attachments/assets/795f237b-cf03-4a40-b2cf-9e41e8e5fccb)

I've updated the model to:
```
bedrock('anthropic.claude-3-5-sonnet-20240620-v1:0')
```

Navigated to the ai-core folder and ran the code with
```
pnpm tsx src/generate-text/amazon-bedrock-tool-call.ts
```

And got:
![image](https://github.com/user-attachments/assets/84c11e93-3b97-49a1-a310-a0cf9422def2)

**Comment by lgrammel** [comment]
Confirmed, it works w/ Haiku but breaks with sonnet.

**Comment by lgrammel** [comment]
Since Haiku works, the issue is most likely on the Amazon side.

**Comment by lgrammel** [comment]
@Amorim33 can you share the `@ai-sdk/bedrock` version that is working with sonnet & tool calls?

**Comment by Amorim33** [comment]
Hmmm I see, it seems likely that this issue could be due to a breaking change on the Amazon's side. The version 0.0.30 of @ai-sdk/amazon-bedrock works correctly with tool calling in my setup, maybe downgrading the runtime version might be a potential workaround.

**Comment by lgrammel** [comment]
Okay seeing the tool calls now, so this has to be an AI SDK issue.

**Comment by Amorim33** [comment]
```
{
  toolResults: [
    {
      toolCallId: 'tooluse_Ktq0iD51T62qIMHCHxgzEQ',
      toolName: 'changeStateToFillTravelersInformation',
      args: {},
      result: [Array]
    }
  ],
  usage: { completionTokens: 152, promptTokens: 1579, totalTokens: 1731 },
  toolCalls: [
    {
      type: 'tool-call',
      toolCallId: 'tooluse_Ktq0iD51T62qIMHCHxgzEQ',
      toolName: 'changeStateToFillTravelersInformation',
      args: {}
    }
  ],
  finishReason: 'tool-calls'
}
```

Tool call results in my application, running with
```
    "@ai-sdk/amazon-bedrock": "0.0.30",
    "ai": "3.4.27",
```

**Comment by lgrammel** [comment]
Found the bug, will prepare fix.

**Comment by Amorim33** [comment]
@lgrammel nice!!

I'm totally available to help work in the fix—I'd actually love the opportunity! Please let me know if there's anything specific I can do to assist.

**Comment by lgrammel** [comment]
You helped a lot with debugging! Fix: https://github.com/vercel/ai/pull/3448

**Comment by Amorim33** [comment]
Awesome pull request!!
Thanks for your attention!

**Comment by lgrammel** [comment]
Fixed in `@ai-sdk/amazon-bedrock@0.0.34`

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Bug**  
Confidence: **High**

### Rationale
This is a confirmed regression in existing Amazon Bedrock tool-calling behavior, subsequently fixed by the maintainers.

### Evidence
- Tool calling worked with @ai-sdk/amazon-bedrock@0.0.30 but failed with 0.0.32 for Claude 3.5 Sonnet, returning empty toolCalls and toolResults.
- The maintainer reproduced the version-specific regression, stated it was an AI SDK issue, and found the bug.
- A fix was merged in PR #3448 and released in @ai-sdk/amazon-bedrock@0.0.34.
